public/app/features/teams/state/actions.ts

```diff
@@ -43,6 +44,20 @@ export function loadTeams(initial = false): ThunkResult<void> {
   };
 }
 
+export function loadUsersTeams(): ThunkResult<void> {
+  return async (dispatch) => {
+    // Early return if the user cannot list teams
+    if (!contextSrv.hasPermission(AccessControlAction.ActionTeamsRead)) {
+      let noTeams: Team[] = [];
+      dispatch(userTeamsLoaded(noTeams));
+      return;
+    }
+
+    const response = await getBackendSrv().get('/api/user/teams');
+    dispatch(userTeamsLoaded({ ...response }));
```

Comment:
```suggestion
    dispatch(userTeamsLoaded(response));
```
There's no need to transform the response array into an object.